### PR TITLE
Support for shape argument in *_like functions

### PR DESCRIPTION
**Pull request by pentschev**
Add support for overriding the shape of the input array when calling `empty_like()`, as recently added to NumPy API in https://github.com/numpy/numpy/pull/13046, starting from NumPy 1.17.

### CODE REVIEW

File: cupy/creation/basic.py [hunk 1]
```diff
@@ -134,7 +145,7 @@ def ones(shape, dtype=float):
     return a
 
 
-def ones_like(a, dtype=None, order='K'):
+def ones_like(a, dtype=None, order='K', shape=None):
```
Comment (toslunar) [inline cupy/creation/basic.py]: To align with the signature `numpy.ones_like(a, dtype=None, order='K', subok=True, shape=None)`, could you insert the `subok` argument?
See e.g. https://github.com/cupy/cupy/blob/fbb745edfad25fa55628ca3571a74781cfb341f2/cupy/core/core.pyx#L351-L352

File: cupy/creation/basic.py [hunk 2]
```diff
@@ -181,7 +198,7 @@ def zeros(shape, dtype=float, order='C'):
     return a
 
 
-def zeros_like(a, dtype=None, order='K'):
+def zeros_like(a, dtype=None, order='K', shape=None):
```
Comment (toslunar) [inline cupy/creation/basic.py]: ditto

File: cupy/creation/basic.py [hunk 3]
```diff
@@ -235,7 +258,7 @@ def full(shape, fill_value, dtype=None):
     return a
 
 
-def full_like(a, fill_value, dtype=None, order='K'):
+def full_like(a, fill_value, dtype=None, order='K', shape=None):
```
Comment (toslunar) [inline cupy/creation/basic.py]: ditto

File: cupy/creation/basic.py (hunk 1 above)
Comment (pentschev) [reply cupy/creation/basic.py]: Sure, will do it.

File: tests/cupy_tests/creation_tests/test_basic.py [hunk 4]
```diff
@@ -231,3 +231,115 @@ def test_full_default_dtype_cpu_input(self, xp, dtype):
...
+
+
+@testing.parameterize(
+    *testing.product({
+        'shape': [(4, 2), (4, 2, 3), (5, 4, 2, 3)],
+    })
+)
+@testing.gpu
+class TestBasicReshape(unittest.TestCase):
+    @testing.with_requires('numpy>=1.17.0')
```
Comment (kmaehashi) [inline tests/cupy_tests/creation_tests/test_basic.py]: Please add test cases that does not depend on NumPy 1.17.0.

File: cupy/creation/basic.py (hunk 1 above)
Comment (pentschev) [reply cupy/creation/basic.py]: Done.

File: tests/cupy_tests/creation_tests/test_basic.py (hunk 4 above)
Comment (pentschev) [reply tests/cupy_tests/creation_tests/test_basic.py]: Done, I added new tests to cover about the same cases, only without comparing to NumPy results.

**Comment by pentschev** [comment]
All change requests have been addressed.

Please note I had to retrigger the build, due to AppVeyor timing out when downloading NumPy. For future reference, is there a way to trigger another build without pushing a commit?

**Comment by kmaehashi** [comment]
> Please note I had to retrigger the build, due to AppVeyor timing out when downloading NumPy. For future reference, is there a way to trigger another build without pushing a commit?

Unfortunately contributors cannot retrigger the test.
Reviewers will take care of it, so please avoid pushing dummy commit to keep git history clean :)

**Comment by kmaehashi** [comment]
This PR implements the spec of unreleased version of NumPy. We agreed to include this into the master branch. This PR can be backported to the stable branch (v6) once after the corresponding version of NumPy has been released.

**Comment by pentschev** [comment]
> Unfortunately contributors cannot retrigger the test.
> Reviewers will take care of it, so please avoid pushing dummy commit to keep git history clean :)

Thanks for letting me know, and sorry for jumping in and pushing that commit unnecessarily.

File: cupy/creation/basic.py [hunk 5]
```diff
@@ -55,6 +62,11 @@ def empty_like(a, dtype=None, order='K'):
             result. 'C' means C-order, 'F' means F-order, 'A' means 'F' if
             ``a`` is Fortran contiguous, 'C' otherwise. 'K' means match the
             layout of ``a`` as closely as possible.
+        subok: Not supported yet, must be None.
+        shape (tuple of ints): Overrides the shape of the result. If order='K'
```
Comment (kmaehashi) [inline cupy/creation/basic.py]: Should be `int or tuple of ints`.

Comment (pentschev) [reply cupy/creation/basic.py]: Sorry for the delay. You're right, I had that copied from the docs of `empty` (and similar) that don't include that information. I pushed a new commit to add that missing information to docs of all those functions now.

**Comment by pentschev** [comment]
Looks like the failure was due to a network issue, could you retrigger @kmaehashi ?

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by pentschev** [comment]
Is there anything else to do here, or could this be merged now that NumPy 1.17 is out?

**Comment by jakirkham** [comment]
@kmaehashi, is this ok to go in or do you need something else from us?

**Comment by pentschev** [comment]
I think 6.3.0 should be release soon, right? It would be great if we could have this merged before that!

**Pull request merged**